File: 0435-non-overlapping-intervals/0435-non-overlapping-intervals.cpp
bool compare(vector<int> a, vector<int> b) {
    return a[1] < b[1];
}

class Solution {
public:
    int eraseOverlapIntervals(vector<vector<int>>& intervals) {
        // sort based on end value
        sort(intervals.begin(), intervals.end(), compare);
        // remove greedily
        int prevEnd = intervals[0][1];
        int deleted = 0;
        for(int i=1; i<intervals.size(); i++) {
            // starting point of current interval should be greater than or equal to the end point of previous element for it to be non-overlapping
            if(intervals[i][0] >= prevEnd) {
                prevEnd = intervals[i][1];
            }
            else {
                deleted++;
            }
        }
        return deleted;
    }
};

File: 0435-non-overlapping-intervals/0435-non-overlapping-intervals.java
class Solution {
    public int eraseOverlapIntervals(int[][] intervals) {
        int n = intervals.length;
        Arrays.sort(intervals, (a, b) -> Integer.compare(a[1], b[1]));

        int ans = 0;
        int currEnd = intervals[0][1];
        for(int i=1; i<n; i++) {
            if(intervals[i][0] >= currEnd) {
                currEnd = intervals[i][1];
            }
            else ans++;
        }

        return ans;
    }
}